study1/week4/max-area-of-island.java

```diff
@@ -0,0 +1,44 @@
+//https://leetcode.com/problems/max-area-of-island/description/
+
+class Solution {
+    public int maxAreaOfIsland(int[][] grid) {
+        //time complexity is O(n*m);
+        //space complexity is O(n*m), n*m times call stacks
+        int n = grid.length;
+        int m = grid[0].length;
+
+        int area = 0;
+
+        for (int y = 0; y < n; y++) {
+            for (int x = 0; x < m; x++) {
+                if (grid[y][x] == 1) {
+                    area = Math.max(area, calArea(grid, x, y));
+                }
+            }
+        }
+        
+        return area;
+    }
+
+    private int calArea(int[][] grid, int x, int y) {
+        if (!isInRange(grid.length, grid[0].length, x, y) || grid[y][x] == 0) {
```

Comment:
Same comment from the floodFill implementation, there's unnecessary comparison here, but I also don't think you can remove this without making this a non-recursive implementation, so it's fine for now.